internal/service/advancedcluster/model_advanced_cluster.go

```diff
@@ -0,0 +1,349 @@
+package advancedcluster
+
+import (
+	"context"
+
+	matlas "go.mongodb.org/atlas/mongodbatlas"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/conversion"
+)
+
+type TfLabelModel struct {
+	Key   types.String `tfsdk:"key"`
+	Value types.String `tfsdk:"value"`
+}
+
+var TfLabelType = types.ObjectType{AttrTypes: map[string]attr.Type{
+	"key":   types.StringType,
+	"value": types.StringType,
+}}
+
+type TfTagModel struct {
+	Key   types.String `tfsdk:"key"`
+	Value types.String `tfsdk:"value"`
+}
+
+var TfTagType = types.ObjectType{AttrTypes: map[string]attr.Type{
+	"key":   types.StringType,
+	"value": types.StringType,
+}}
+
+type TfBiConnectorConfigModel struct {
+	ReadPreference types.String `tfsdk:"read_preference"`
+	Enabled        types.Bool   `tfsdk:"enabled"`
+}
+
+var TfBiConnectorConfigType = types.ObjectType{AttrTypes: map[string]attr.Type{
+	"read_preference": types.StringType,
+	"enabled":         types.BoolType,
+}}
+
+type TfAdvancedConfigurationModel struct {
+	DefaultReadConcern               types.String `tfsdk:"default_read_concern"`
+	DefaultWriteConcern              types.String `tfsdk:"default_write_concern"`
+	MinimumEnabledTLSProtocol        types.String `tfsdk:"minimum_enabled_tls_protocol"`
+	OplogSizeMB                      types.Int64  `tfsdk:"oplog_size_mb"`
+	OplogMinRetentionHours           types.Int64  `tfsdk:"oplog_min_retention_hours"`
+	SampleSizeBiConnector            types.Int64  `tfsdk:"sample_size_bi_connector"`
+	SampleRefreshIntervalBiConnector types.Int64  `tfsdk:"sample_refresh_interval_bi_connector"`
+	TransactionLifetimeLimitSeconds  types.Int64  `tfsdk:"transaction_lifetime_limit_seconds"`
+	FailIndexKeyTooLong              types.Bool   `tfsdk:"fail_index_key_too_long"`
+	JavascriptEnabled                types.Bool   `tfsdk:"javascript_enabled"`
+	NoTableScan                      types.Bool   `tfsdk:"no_table_scan"`
+}
+
+var tfAdvancedConfigurationType = types.ObjectType{AttrTypes: map[string]attr.Type{
+	"default_read_concern":                 types.StringType,
+	"default_write_concern":                types.StringType,
+	"minimum_enabled_tls_protocol":         types.StringType,
+	"oplog_size_mb":                        types.Int64Type,
+	"oplog_min_retention_hours":            types.Int64Type,
+	"sample_size_bi_connector":             types.Int64Type,
+	"sample_refresh_interval_bi_connector": types.Int64Type,
+	"transaction_lifetime_limit_seconds":   types.Int64Type,
+	"fail_index_key_too_long":              types.BoolType,
+	"javascript_enabled":                   types.BoolType,
+	"no_table_scan":                        types.BoolType,
+}}
+
+type tfConnectionStringModel struct {
```

Comment:
some models/types are public (upercase) while others are private (lowercase). Would unify, usually we are keeping them private but when adding unit testing the have to be made public.